lib/collection/src/shards/replica_set/partial_snapshot_meta.rs

```diff
@@ -55,4 +68,51 @@ impl PartialSnapshotMeta {
                 backtrace: None,
             })
     }
+
+    pub fn recovery_timestamp(&self) -> u64 {
+        self.recovery_timestamp.load(atomic::Ordering::Relaxed)
+    }
+
+    pub fn snapshot_recovered(&self) {
+        let timestamp = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap_or(Duration::ZERO)
+            .as_secs();
+
+        self.recovery_timestamp
+            .store(timestamp, atomic::Ordering::Relaxed);
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct RequestTracker {
+    requests: Arc<AtomicUsize>,
```

Comment:
We _can_ also use [`Arc`](https://doc.rust-lang.org/std/sync/struct.Arc.html) for this, give out weak references, and count using [`weak_count`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.weak_count). This way we don't reinvent the wheel and don't make mistakes.

Reply:
Valid idea, but it's more obscure than explicit atomic, IMO.

Reply:
Fair enough. I'd also consider using Arc here more of a hack.